src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let finalString = '';
+    let newSeparator = separator;
+    let counter = 0;
```

Comment:
why do you need this counter? if you want to check index of char use a common `for` loop with increment